src/stapi_fastapi/routers/root_router.py

```diff
@@ -140,21 +140,51 @@ def get_root(self, request: Request) -> RootResponse:
     def get_conformance(self, request: Request) -> Conformance:
         return Conformance(conforms_to=self.conformances)
 
-    def get_products(self, request: Request) -> ProductsCollection:
-        return ProductsCollection(
-            products=[pr.get_product(request) for pr in self.product_routers.values()],
-            links=[
+    def get_products(
+        self, request: Request, next: str | None = None, limit: int = 10
+    ) -> ProductsCollection:
+        start = 0
+        product_ids = [*self.product_routers.keys()]
+        try:
```

Comment:
I don't see how this block of code can ever work as seems to be intended. `next` is a str, and should always cause `product_ids.index(next)` to raise `ValueError`